src/App.tsx

```diff
@@ -15,49 +15,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  NAME = 'name',
+  LENGTH = 'length',
+}
+
+const getPreparedGoods = (
+  goodsFrom: string[],
+  sortField: SortField | '',
+  reverse: boolean,
+): string[] => {
+  const goods = [...goodsFrom];
+
+  if (sortField) {
+    goods.sort((a, b) => {
+      switch (sortField) {
+        case SortField.NAME:
+          return a.localeCompare(b);
+
+        case SortField.LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    goods.reverse();
+  }
+
+  return goods;
+};
+
+function getReset(sortField: SortField | '', reverse: boolean): boolean {
+  return sortField !== '' || reverse;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortField | ''>('');
+  const [reverse, setReversed] = useState<boolean>(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
+  const reset = getReset(sortField, reverse);
```

Comment:
this variable looks redundant since you can just display reset button if `sortBy !== '' || isReversed` directly at jsx